src/librustc_codegen_llvm/mono_item.rs

```diff
@@ -136,7 +136,39 @@ fn predefine_static<'a, 'tcx>(cx: &CodegenCx<'a, 'tcx>,
     let ty = instance.ty(cx.tcx);
     let llty = cx.layout_of(ty).llvm_type(cx);
 
-    let g = declare::define_global(cx, symbol_name, llty).unwrap_or_else(|| {
+    let addr_space = match cx.tcx.hir.get_if_local(def_id) {
+        Some(hir::map::NodeItem(&hir::Item {
+            node: hir::ItemStatic(_, hir::MutMutable, _), ..
+        })) |
+        Some(hir::map::NodeForeignItem(&hir::ForeignItem {
+            node: hir::ForeignItemStatic(_, true), ..
+        })) => cx.default_global_addr_space(),
+
+        Some(hir::map::NodeItem(&hir::Item {
+            node: hir::ItemStatic(..), ..
+        })) |
+        Some(hir::map::NodeForeignItem(&hir::ForeignItem {
+            node: hir::ForeignItemStatic(..), ..
+        })) => {
+            if cx.type_is_freeze(ty) {
+                cx.default_const_addr_space()
+            } else {
+                cx.default_global_addr_space()
+            }
+        },
```

Comment:
We should avoid touching the HIR here, but it's not clear what requirements there actually are.

Reply:
We need the address space of the global for the global's LLVM type, meaning it needs to compute the same address space as `consts::codegen_static`.

At the time, I wasn't aware of a better way, but the next version now does what `MonoItemExt::define` does.